optimum/neuron/generation/utils.py

```diff
@@ -1049,15 +632,337 @@ def generate(
+                The id of the *end-of-sequence* token. Optionally, use a list to set multiple *end-of-sequence* tokens.
+            output_attentions (`bool`, *optional*, defaults to `False`):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more details.
+            output_hidden_states (`bool`, *optional*, defaults to `False`):
+                Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors
+                for more details.
+            output_scores (`bool`, *optional*, defaults to `False`):
+                Whether or not to return the prediction scores. See `scores` under returned tensors for more details.
+            return_dict_in_generate (`bool`, *optional*, defaults to `False`):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+            synced_gpus (`bool`, *optional*, defaults to `False`):
+                Whether to continue running the while loop until max_length (needed for ZeRO stage 3)
+            seq_length (`Optional[int]`, defaults to `False`):
+                Length of current input_ids sequence
+            is_traced_inference (`bool`, defaults to `False`):
+                Whether the decoder is traced or using XLA lazy tensor. If the decoder is traced, next tokens and the beam scores
+                are computed inside the decoder.
+            model_kwargs:
+                Additional model specific keyword arguments will be forwarded to the `forward` function of the model.
+                If model is an encoder-decoder model the kwargs should include `encoder_outputs`.
+
+        Return:
+            [`~generation.GreedySearchDecoderOnlyOutput`], [`~generation.GreedySearchEncoderDecoderOutput`] or
+            `torch.LongTensor`: A `torch.LongTensor` containing the generated tokens (default behaviour) or a
+            [`~generation.GreedySearchDecoderOnlyOutput`] if `model.config.is_encoder_decoder=False` and
+            `return_dict_in_generate=True` or a [`~generation.GreedySearchEncoderDecoderOutput`] if
+            `model.config.is_encoder_decoder=True`.
+
+        Examples:
+
+        ```python
+        >>> from transformers import AutoTokenizer
+        >>> from optimum.neuron import NeuronModelForSeq2SeqLM
+
+        >>> tokenizer = AutoTokenizer.from_pretrained("t5-small")
+        >>> input_shapes = {"batch_size": 1, "sequence_length": 128, "num_beams": 1}
+        >>> model = NeuronModelForSeq2SeqLM.from_pretrained("t5-small", export=True, dynamic_batch_size=False, **input_shapes)
+
+        >>> input_prompt = "translate English to German: Lets eat good food."
+        >>> inputs = tokenizer(input_prompt, return_tensors="pt")
+
+        >>> outputs = model.greedy_search(input_ids)
+
+        >>> results = [tokenizer.decode(t, skip_special_tokens=True) for t in outputs]
+        ```
+        """
```

Comment:
I dont know if we should copy/paste all the docstring from functions we override. 
I am personally not too much in favor of doing so because it adds maintainance on our side for not so much added value. But it's just a thought.

Reply:
I want to add the explanation for `is_traced_inference`, felt it would be better to have it completed, especially we will tailor quite a lot of those args to be compatible with neuron in near future, it would be better to have a complete docstring explaining that.